### PR TITLE
Work on FAR/Rename w/UI reliablity with cancellation

### PR DESCRIPTION
**Pull request by Colengms**

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -674,47 +689,63 @@ export class DefaultClient implements Client {
...
+                                        newName: newName,
+                                        position: Position.create(position.line, position.character),
+                                        textDocument: this.client.languageClient.code2ProtocolConverter.asTextDocumentIdentifier(document)
+                                    };
+                                    referencesParams = params;
+                                    this.client.notifyWhenReady(() => {
+                                        // The current request is represented by referencesParams.  If a request detects
+                                        // referencesParams does not match the object used when creating the request, abort it.
+                                        if (params !== referencesParams) {
+                                            reject();
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts:708]: `reject` does not return.  So unless you want to run the rest of the code in this function after rejecting, you should early return here.

Comment (Colengms) [reply Extension/src/LanguageServer/client.ts:708]: I do want it to run the rest of the code in that function.

**Comment by michelleangela** [comment]
Does this address https://github.com/microsoft/vscode-cpptools/issues/4052 ?

**Comment by Colengms** [comment]
> Does this address #4052 ?

I don't believe so.

**Review by michelleangela — APPROVED** [review]

File: Extension/src/LanguageServer/client.ts (hunk 1 above)
Comment (Colengms) [reply Extension/src/LanguageServer/client.ts:708]: Ack, you were correct, this one should have returned.

**Review by bobbrow — APPROVED** [review]

**Pull request merged**